pkg/cli/admin/catalog/mirror.go

```diff
@@ -0,0 +1,280 @@
+			return "", err
+		}
+		if err := e.Validate(); err != nil {
+			return "", err
+		}
+		if err := e.Run(); err != nil {
+			return "", err
+		}
+		if len(e.Mappings) < 1 {
+			return "", fmt.Errorf("couldn't extract database")
+		}
+		klog.Infof("wrote database to %s", path.Join(e.Mappings[0].To, "bundles.db"))
+		return path.Join(e.Mappings[0].To, "bundles.db"), nil
+	}
+	o.DatabaseExtractor = extractor
+	return nil
+}
+
+func (o *MirrorCatalogOptions) Validate() error {
+	if o.DatabasePath == "" {
+		return fmt.Errorf("must specify path for database")
+	}
+	if o.ManifestDir == "" {
+		return fmt.Errorf("must specify path for manifests")
+	}
+	return nil
+}
+
+func (o *MirrorCatalogOptions) Run() error {
+	indexMirrorer, err := mirror.NewIndexImageMirror(o.IndexImageMirrorerOptions.ToOption(),
+		mirror.WithSource(o.SourceRef.String()),
+		mirror.WithDest(o.Dest),
+	)
+	if err != nil {
+		return err
+	}
+	mapping, err := indexMirrorer.Mirror()
+	if err != nil {
+		klog.Warningf("errors during mirroring. the full contents of the catalog may not have been mirrored: %s", err.Error())
+	}
+
+	return WriteManifests(o.SourceRef.Name, o.ManifestDir, mapping)
+}
+
+func WriteManifests(name, dir string, mapping map[string]string) error {
+	f, err := os.Create(path.Join(dir, "mapping.txt"))
+	if err != nil {
+		return err
+	}
+	defer func() {
+		if err := f.Close(); err != nil {
+			klog.Warningf("error closing file")
+		}
+	}()
+
+	icsp := operatorv1alpha1.ImageContentSourcePolicy{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: operatorv1alpha1.GroupVersion.String(),
+			Kind:       "ImageContentSourcePolicy"},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+		Spec: operatorv1alpha1.ImageContentSourcePolicySpec{
+			RepositoryDigestMirrors: []operatorv1alpha1.RepositoryDigestMirrors{},
+		},
+	}
+
+	for k, v := range mapping {
+		fromRef, err := reference.Parse(k)
+		if err != nil {
+			klog.Warningf("error parsing ref for %s", k)
+			continue
+		}
+		toRef, err := reference.Parse(v)
+		if err != nil {
+			klog.Warningf("error parsing ref for %s", v)
+			continue
+		}
+		icsp.Spec.RepositoryDigestMirrors = append(icsp.Spec.RepositoryDigestMirrors, operatorv1alpha1.RepositoryDigestMirrors{
+			Source:  fromRef.AsRepository().String(),
+			Mirrors: []string{toRef.AsRepository().String()},
+		})
+
+		// omit digest from target if digest exists
+		to := v
+		if toRef.ID != "" {
+			to = toRef.AsRepository().String()
+		}
+		if _, err := f.WriteString(fmt.Sprintf("%s=%s\n", k, to)); err != nil {
+			return err
+		}
+	}
+
+	// Create an unstructured object for removing creationTimestamp
+	unstructuredObj := unstructured.Unstructured{}
+	unstructuredObj.Object, err = runtime.DefaultUnstructuredConverter.ToUnstructured(&icsp)
+	if err != nil {
+		return fmt.Errorf("ToUnstructured error: %v", err)
+	}
+	delete(unstructuredObj.Object["metadata"].(map[string]interface{}), "creationTimestamp")
```

Comment:
I see this code copied from https://github.com/openshift/oc/blob/master/pkg/cli/admin/release/mirror.go and I'm not that happy about it :cry: but I'll let you land it. Please fix this in a followup.
Since you have the access to the resource just set the creation timestamp to nil and for printing how about using printers? You can force it to produce yaml, always.